cli/src/main/scala/co/topl/brambl/cli/controllers/WalletController.scala

```diff
@@ -35,6 +42,84 @@ class WalletController[F[_]: Sync](
     genusQueryAlgebra: dataApi.GenusQueryAlgebra[F]
 ) {
 
+  def addSecret(
+      secretTxt: String,
+      digest: DigestType
+  ): F[Either[String, String]] = {
+    import co.topl.crypto.hash.implicits.sha256Hash
+    import cats.implicits._
+    val paddedSecret = secretTxt.getBytes() ++ Array
+      .fill(secretTxt.getBytes().length - 32)(0.toByte)
+    val hashedSecret =
+      if (digest == Sha256)
+        sha256Hash.hash(
+            paddedSecret
+          ).value
+      else
+        (new Blake2b256).hash(
+          paddedSecret
+        )
+    val propDigest = Proposition.Digest(
+      digest.digestIdentifier,
+      Digest(
+        ByteString.copyFrom(
+          hashedSecret
+        )
+      )
+    )
+    for {
+      somePreimage <- walletStateAlgebra.getPreimage(
+        propDigest
+      )
+      res <-
+        if (somePreimage.isEmpty)
+          walletStateAlgebra
+            .addPreimage(
+              Preimage(
+                ByteString.copyFrom(secretTxt.getBytes()),
+                ByteString.copyFrom(
+                  Array.fill(secretTxt.getBytes().length - 32)(0.toByte)
```

Comment:
((same as above comment))

```suggestion
                  Array.fill(32 - secretTxt.getBytes().length)(0.toByte)
```